synapse/api/errors.py

```diff
@@ -147,6 +147,14 @@ def error_dict(self):
         return cs_error(self.msg, self.errcode)
 
 
+class InvalidAPICallError(SynapseError):
```

Comment:
I feel like if we're going to introduce a new subclass here we should consider using it more widely. (Or at least file an issue about it.)

Reply:
That's fair. I had in mind #8445 here---e.g. that any systematic approach to validation could make use of this subclass.

Reply:
Yeah that makes sense as doing it as part of that issue!